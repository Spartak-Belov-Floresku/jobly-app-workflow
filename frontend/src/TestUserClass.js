import GlobalContext from "./helper/GlobalContext";

const mockUser = {
  username: "testuser",
  first_name: "testfirst",
  last_name: "testlast",
  email: "test@test.net",
  photo_url: null,
};

const TestUserClass = ({ children, currUser = mockUser, jobHasBeenApplied = () => false }) => (
  <GlobalContext.Provider value={{ currUser, jobHasBeenApplied }}>
    {children}
  </GlobalContext.Provider>
);

export { TestUserClass };