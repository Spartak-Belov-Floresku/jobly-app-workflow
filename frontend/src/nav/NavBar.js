import "./NavBar.css";
import { useContext } from "react";
import { NavLink } from "react-router-dom";
import { Navbar, Nav, NavItem } from "reactstrap";
import GlobalContext from "../helper/GlobalContext";

/** Navigation bar for site. Shows up on every page.
 *
 * When user is logged in, shows links to main areas of site. When not,
 * shows link to Login and Signup forms.
 *
 * Rendered by App.
 */

const NavBar = ({ Logout }) => {
  
  const { currUser } = useContext(GlobalContext);

  const loggedInNav = () => {
    return (
        <div>
          <Navbar expand="md">
            <NavLink to="/" className="navbar-brand">
              Jobly
            </NavLink>
            <Nav className="ml-auto" navbar>
              <NavItem>
                <NavLink to="/companies">Companies</NavLink>
              </NavItem>
              <NavItem>
                <NavLink to="/jobs">Jobs</NavLink>
              </NavItem>
              <NavItem>
                <NavLink to="/profile">Profile</NavLink>
              </NavItem>
              <NavItem>
                <NavLink to="/login" onClick={Logout}>Log out: <b className="username">{currUser.username}</b></NavLink>
              </NavItem>
            </Nav>
          </Navbar>
        </div>
    );
  }

  const loggedOutNav = () => {
    return (
        <div>
          <Navbar expand="md">
            <NavLink to="/" className="navbar-brand">
              Jobly
            </NavLink>
            <Nav className="ml-auto" navbar>
              <NavItem>
                <NavLink to="/login">Login</NavLink>
              </NavItem>
              <NavItem>
                <NavLink to="/signup">Sing Up</NavLink>
              </NavItem>
            </Nav>
          </Navbar>
        </div>
    );
  }

  return (currUser ? loggedInNav() : loggedOutNav());
}

export default NavBar;