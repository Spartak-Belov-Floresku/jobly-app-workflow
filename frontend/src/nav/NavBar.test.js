import { render } from "@testing-library/react";
import { MemoryRouter } from "react-router";
import NavBar from "./NavBar";
import { TestUserClass } from "../TestUserClass";

it("renders without crashing logged in mock user", () => {
  render(
      <MemoryRouter>
        <TestUserClass>
          <NavBar />
        </TestUserClass>
      </MemoryRouter>,
  );
});

it("matches snapshot logged in mock user", () => {
  const { asFragment } = render(
    <MemoryRouter>
      <TestUserClass>
        <NavBar />
      </TestUserClass>
    </MemoryRouter>,
  );
  expect(asFragment()).toMatchSnapshot();
});

  it("matches snapshot when logged out", () =>{
    const { asFragment } = render(
      <MemoryRouter>
        <TestUserClass currentUser={null}>
          <NavBar />
        </TestUserClass>
      </MemoryRouter>,
    );

    expect(asFragment()).toMatchSnapshot();
});