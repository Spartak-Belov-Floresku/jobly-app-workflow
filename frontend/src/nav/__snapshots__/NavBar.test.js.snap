// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot logged in mock user 1`] = `
<DocumentFragment>
  <div>
    <nav
      class="navbar navbar-expand-md"
    >
      <div
        class="container-fluid"
      >
        <a
          aria-current="page"
          class="navbar-brand active"
          href="/"
        >
          Jobly
        </a>
        <ul
          class="ml-auto navbar-nav"
        >
          <li
            class="nav-item"
          >
            <a
              href="/companies"
            >
              Companies
            </a>
          </li>
          <li
            class="nav-item"
          >
            <a
              href="/jobs"
            >
              Jobs
            </a>
          </li>
          <li
            class="nav-item"
          >
            <a
              href="/profile"
            >
              Profile
            </a>
          </li>
          <li
            class="nav-item"
          >
            <a
              href="/login"
            >
              Log out: 
              <b
                class="username"
              >
                testuser
              </b>
            </a>
          </li>
        </ul>
      </div>
    </nav>
  </div>
</DocumentFragment>
`;

exports[`matches snapshot when logged out 1`] = `
<DocumentFragment>
  <div>
    <nav
      class="navbar navbar-expand-md"
    >
      <div
        class="container-fluid"
      >
        <a
          aria-current="page"
          class="navbar-brand active"
          href="/"
        >
          Jobly
        </a>
        <ul
          class="ml-auto navbar-nav"
        >
          <li
            class="nav-item"
          >
            <a
              href="/companies"
            >
              Companies
            </a>
          </li>
          <li
            class="nav-item"
          >
            <a
              href="/jobs"
            >
              Jobs
            </a>
          </li>
          <li
            class="nav-item"
          >
            <a
              href="/profile"
            >
              Profile
            </a>
          </li>
          <li
            class="nav-item"
          >
            <a
              href="/login"
            >
              Log out: 
              <b
                class="username"
              >
                testuser
              </b>
            </a>
          </li>
        </ul>
      </div>
    </nav>
  </div>
</DocumentFragment>
`;
